Typography: Added basic styles for paragraphs and lists

thejdeep opened the pull request.

geekman-rohit commented:
@thejdeep please sign our CLA at  http://contribute.jquery.org/CLA/

thejdeep commented:
Done. I am waiting for it to re-check.

thejdeep commented:
@geekman-rohit Shouldn't be taking this long ?

geekman-rohit commented:
@thejdeep the bot doesn't update automatically, it will update the next time you push a commit . 

If its time to pull in ur PR and the label is not updated, I'll confirm with the team that you have signed and update the label manually.

arschmitz commented:
@geekman-rohit the CLA does recheck automatically once a minute or so this is an actual failure.

@thejdeep the problem is you need to sign with your full name ( which i see you did at first ) but the name in your git config is just your first name and thats what its matching and checking against. It finds a match but fails on it only having first name. It should say this in the error details but looks like its returning a 404 on the link sorry about that!

thejdeep committed:
Typography: Added styles for paragraphs and lists

Typography: Eliminated need of a mixin for p

Typography: Eliminated need of mixin for p

Typography: Shorthand margins

Typography: Eliminated need of mixin for p

Typography: Fixed indentation

Typography: Fixed default fontSize

Typography: Changed to follow naming conventions

geekman-rohit commented:
@thejdeep  you'll have to pull in latest commits and rebase  to resolve conflicts

geekman-rohit commented on scss/atoms/typography/_mixins.scss:
```diff
@@ -1,5 +1,18 @@
+
+@mixin paragraph($style) {
+	text-align: map-get($style,textAlignment);
+	margin: em(map-get($style,margin)) 0;
```
This mixin is called only once, do we really need it as a mixin?

geekman-rohit commented on scss/atoms/typography/_typography.scss:
```diff
@@ -21,40 +21,43 @@ body {
+ol ul,
+ul ol,
+ol ol {
+  margin-bottom: 0;
```
Indentation wrong

geekman-rohit commented on scss/atoms/typography/_typography.scss:
```diff
@@ -21,40 +21,43 @@ body {
+}
+
+p {
+	@include paragraph($p);
```
check my comment above

geekman-rohit commented on .gitignore:
```diff
@@ -10,3 +10,5 @@ external/
 icons/svg-min/
 .sass-cache/
 dist/
+.DS_Store
```
Debatable. These files are operating system specific. shouldn't be part of the repo's gitignore. Atleast thats what I think. @arthurvr  opinion?

sfrisk replied:
Remove this from .gitignore.   Reasoning in the discussion around #108

sfrisk commented on scss/atoms/typography/_typography.scss:
```diff
@@ -21,40 +21,46 @@ body {
+
+p {
+	text-align: map-get($p,textAlignment);
+	margin: em(map-get($p,margin)) 0;
```
have both values for margin be in the variable, not just the top/bottom.  Otherwise people have no way to control the left/right margin.

sfrisk commented on scss/variables/typography.js:
```diff
@@ -52,7 +53,113 @@ chassis.typography = {
+		value: {
+			margin: "25px",
+			fontSize: "16px",
+			textAlignment: "justify"
```
Not sure I would go with justify here. for the basic <p> styling.

sfrisk commented on scss/atoms/typography/_typography.scss:
```diff
@@ -21,40 +21,46 @@ body {
+
+ul, ol {
+	margin-top: 0;
+	margin-bottom: 10px;
```
Shorthand margins

sfrisk commented on scss/atoms/typography/_typography.scss:
```diff
@@ -21,40 +21,46 @@ body {
+ol ul,
+ul ol,
+ol ol {
+  	margin-bottom: 0;
```
Shorthand margins

sfrisk commented on scss/variables/typography.js:
```diff
@@ -52,7 +53,113 @@ chassis.typography = {
+	p: {
+		name: "P style",
+		value: {
+			margin: "25px",
```
Is this margin on all 4 sides?

sfrisk replied:
Looks like your mixin has this only being the top/bottom margin.

geekman-rohit commented on scss/atoms/typography/_typography.scss:
```diff
@@ -21,40 +21,45 @@ body {
+}
+
+ul, ol {
+	margin: 0 0 10px;
```
still something wrong with the indentation. No spaces. Single tab. Turn off soft tabs in your text editor.

geekman-rohit commented on scss/atoms/typography/_typography.scss:
```diff
@@ -21,40 +21,45 @@ body {
+ol ul,
+ul ol,
+ol ol {
+  	margin: 0;
```
still something wrong with the indentation. No spaces. Single tab. Turn off soft tabs in your text editor.

geekman-rohit commented on scss/variables/typography.js:
```diff
@@ -52,7 +53,114 @@ chassis.typography = {
+		value: {
+			marginTopBottom: "25px",
+			marginLeftRight: "0px",
+			fontSize: "16px",
```
The font size should be default. use the fontSize variable. If the default font size is too large or small ,. we can discuss it in a meeting. See my next comment

geekman-rohit replied:
use the font-default:font-size map or the defaultFont variable

geekman-rohit commented on scss/atoms/typography/_typography.scss:
```diff
@@ -21,40 +21,45 @@ body {
+	text-align: map-get($p,textAlignment);
+	margin: em(map-get($p,marginTopBottom)) em(map-get($p,marginLeftRight));
+	line-height: 1.5;
+	font-size: map-get($defaultFont, font-size);
```
$defaultFont doesnot exist anymore. Its $font-default now. 
check https://github.com/jquery/css-chassis/blob/master/scss/variables/typography.js
  Rebase this branch on jquery/master to see. That will also remove conflicts.

geekman-rohit commented on scss/variables/typography.js:
```diff
@@ -52,7 +53,113 @@ chassis.typography = {
+			capitalization: "uppercase"
+		}
+	},
+	p: {
```
as discussed in meeting on tuesday. Double quotes around scss variables defined in jsass 
check my comment on : #135

geekman-rohit commented on scss/variables/typography.js:
```diff
@@ -52,7 +53,113 @@ chassis.typography = {
+	p: {
+		name: "P style",
+		value: {
+			marginTopBottom: "25px",
```
And margin-top and margin-bottom should be different variables probably.  Same goes for marginLeftRight
Also we agreed on Tuesday to hyphenate variables and not camel case them

sfrisk commented:
Is there a visual demo of paragraph and list tags?

geekman-rohit commented on scss/atoms/typography/_typography.scss:
```diff
@@ -44,6 +44,24 @@ h6 {
+	text-align: map-get($p,text-alignment);
+	margin: em(map-get($p,margin-top)) em(map-get($p,margin-right)) em(map-get($p,margin-bottom)) em(map-get($p,margin-left));
+	line-height: 1.5;
+	font-size: map-get($font-default, font-size);
```
You are assigning the default font size to p directly. That will happen anyways as p will inherit the style.
Instead you should have a font-size variable in $p map which is set to the default font size be default, and user can overwrite if he wants

geekman-rohit replied:
use em on this so we get value in em

geekman-rohit commented on scss/variables/typography.js:
```diff
@@ -161,6 +161,16 @@ chassis.typography = {
+			"margin-left": "0px",
+			"margin-right": "0px",
+			"text-alignment": "left"
+		}
```
check my comment above:`" font-size" :fontSize`

thejdeep committed:
Typography: Added font-size to p map

sfrisk commented on scss/atoms/typography/_typography.scss:
```diff
@@ -44,6 +44,24 @@ h6 {
+}
+
+ul, ol {
+	margin: 0 0 10px;
```
margins should probably be put into a variable

sfrisk replied:
also just styling for ol/ul and nested ol/ul?  no <li> styling?

thejdeep commented:
Will work on this over the next few days. Was swamped with finals.t

sfrisk commented:
Any updates?
